### PR TITLE
mean fill bug

### LINKED ISSUE #4514: Issue w/ fill_mean in `ReceptiveField`
From @nbara 

> Isn't there a small bug Line 414 of receptive_field.py ? delayed[...] = fill_value is over-indented isn't it? Meaning the fill_value is effectively only applied when epoch_axis is not None?

I took a look and think this is true...we should fix this!

### PR DESCRIPTION
**Pull request by choldgraf**
I believe this fixes #4514 and adds a test for it...should double check the tests @larsoner since I wrote this while on a flight back from NYC and my brain was a little fried :-)

### CODE REVIEW

File: mne/decoding/tests/test_receptive_field.py [hunk 1]
```diff
@@ -123,6 +123,12 @@ def test_receptive_field():
     rf.fit(X, y)
     assert_array_equal(rf.delays_, np.arange(tmin, tmax + 1))
 
+    # Testing fill mean
+    rf = ReceptiveField(tmin, tmax, 1, feature_names, fill_mean=True,
+                        estimator=mod)
+    rf.fit(X, y)
+    assert_array_equal(rf.delays_, np.arange(tmin, tmax + 1))
```
Comment (larsoner) [inline mne/decoding/tests/test_receptive_field.py]: This doesn't test that `fill_value` actually does anything. Can you add a test that fails in `master` but passes here?

Comment (massich) [reply mne/decoding/tests/test_receptive_field.py]: more over, `fill_value` is not in `ReceptiveField`'s signature [here](https://github.com/mne-tools/mne-python/blob/a22466f326cdc0724b5dbd4ae5e85005193eef14/mne/decoding/receptive_field.py#L91)

Comment (larsoner) [reply mne/decoding/tests/test_receptive_field.py]: `fill_value` isn't really meant to be public

Comment (massich) [reply mne/decoding/tests/test_receptive_field.py]: So this is not referring to `ReceptiveField`'s signature?
img

Comment (larsoner) [reply mne/decoding/tests/test_receptive_field.py]: Yeah you're right, didn't see it was in `__init__` -- it's also not supposed to be publicly exposed, so this is the wrong way to check it

**Comment by larsoner** [comment]
@choldgraf do you want me to take over?

**Comment by choldgraf** [comment]
@larsoner if you've got the time, please go for it...been at neurohackweek for the last week and my to-do list has exploded as a result :-P

**Comment by larsoner** [comment]
Closing for #4543

**Pull request closed without merging**